File: 2015/Day4/Day4.py
import hashlib

hashNum = 0

while(True):
    BHashNum = bytes(str(hashNum), 'utf-8')
    result = hashlib.md5(b"iwrupvqb" + BHashNum)
    DData = result.hexdigest()

    if DData[0:5] == "00000":
        print(hashNum)
        break
    else:
        hashNum += 1
        if (hashNum % 10000 == 0):
            print(hashNum)

File: 2015/Day4/Day4.cpp
#include <iostream>
#include <string>
#include "Poco/MD5Engine.h"
#include "Poco/DigestStream.h"

using namespace std;

int main() {
    Poco::MD5Engine md5;
    string hash = "iwrupvqb", result = "";
    int hashnum = 0;

    while (true)
    {
        md5.update(hash + to_string(hashnum));
        result = Poco::DigestEngine::digestToHex(md5.digest());
        
        if ("000000" == result.substr(0, 6)) {
            cout << hashnum << endl;
            break;
        }
        else {
            hashnum++;
            // if (hashnum % 10000 == 0)
            //     cout << hashnum << endl;
        }
    }
}